### PR TITLE
#168243560 user-signup-api

### PR DESCRIPTION
**Pull request by Jajabenit250**
#### What does this PR do?
User Should Sign Up (API)

#### Description of Task to be completed?
- added Route to navigate link
- adding helper to help in responses and errors
- adding middleware to validate user inputs
- adding models as our data store
- adding test to check if everything works well and configure covellars 
- configure Travis

#### How should this be manually tested?
Clone the repository and cd into it. run npm install and after run npm start. Open postman and run
http://localhost/api/v1/auth/signup  and the add required parameters in a body.

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#168243560

#### Screenshots (if appropriate)
final touches

Questions:

### CODE REVIEW

**Review by oesukam — CHANGES_REQUESTED** [review]
Add the screenshot of your Postman's response

**Comment by Jajabenit250** [comment]
> Add the screenshot of your Postman's response

@oesukam  screenshot added

**Review by oesukam — CHANGES_REQUESTED** [review]
`is_admin` field should also follow the same format as the rest

File: .coveralls.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+service_name: travis-pro
+repo_token: v8USzobOwk9BKjiXaih8UkJ2tevWWLxK5
```
Comment (oesukam) [inline .coveralls.yml]: You can't have your token put in public due to security breach

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+node_js:
+  - "stable"
+install: npm install
+cache:
+  directories:
+    - node_modules
+script:
+  - npm test
+after_success:
+- npm run coveralls
```
Comment (oesukam) [inline .travis.yml]: Blank line required

**Review by oesukam — CHANGES_REQUESTED** [review]
The response status data type is different with the one from the `ADC` document

**Comment by Jajabenit250** [comment]
> The response status data type is different with the one from the `ADC` document

@oesukam changes applied

File: .coveralls.yml (hunk 1 above)
Comment (Jajabenit250) [reply .coveralls.yml]: i removed the files but travis is not working any suggestion ??

**Review by oesukam — CHANGES_REQUESTED** [review]
@Jajabenit250 Did you check the Postman's response? The status's field still doesn't follow the required format

**Review by oesukam — APPROVED** [review]

**Pull request merged**